src/core/lv_obj_draw.c

```diff
@@ -181,14 +180,12 @@ void lv_obj_init_draw_rect_dsc(lv_obj_t * obj, uint32_t part, lv_draw_rect_dsc_t
     }
 #endif
 
-    if(part != LV_PART_MAIN) {
```

Comment:
This check is still needed, to prevent opacity glitches.

Reply:
Created upstream issue:
[https://github.com/lvgl/lvgl/issues/4611](https://github.com/lvgl/lvgl/issues/4611)

Reply:
Associated PR:
[https://github.com/lvgl/lvgl/pull/4617](https://github.com/lvgl/lvgl/pull/4617)